src/test/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheckTest.java

```diff
@@ -288,4 +293,27 @@ public void testCacheHeaderWithoutFile() throws Exception {
 
         verify(checker, getPath("InputHeader.java"), expected);
     }
+
+    @Test
+    public void testIgnoreLinesSorted() throws Exception {
+        final HeaderCheck check = new HeaderCheck();
+        check.setIgnoreLines(5, 9, 9, 11, 2);
+        final int[] ignoreLines = (int[]) Whitebox.getInternalState(check, "ignoreLines");
+        assertArrayEquals(new int[]{2, 5, 9, 9, 11}, ignoreLines);
```

Comment:
We try to avoid pure UT as much as possible. So you might noticed that all our Tests are functional and test Check by real execution.
Can we test `ignoredLines` by providing input file, regexp and recheck that ignored lines affect validation expectations.